### PR TITLE
[elastic-agent] Improve log messages

### PR DESCRIPTION
**Pull request by ruflin**
This PR improves the log messages to make them easier to consume by users.

* Change unroll to unenroll, that is what we use in the UI too
* Trigger unenroll only when the maxCounter is exceeded (nit)
* Move Artifact has been built with security disable to info. This only happens for internal builds
* Move log message around binary already exists to debug level as this is an expected behaviour
* Be more specific about binary downloads
* For state changes, move the message first. That is the most interesting part for a user

### CODE REVIEW

**Review by mtojek — APPROVED** [review]
LGTM

File: x-pack/elastic-agent/pkg/reporter/log/format.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ const (
 
 const (
 	// e.g "2006-01-02T15:04:05: type: 'STATE': event type: 'STARTING' message: Application 'filebeat' is starting."
```
Comment (blakerouse) [inline x-pack/elastic-agent/pkg/reporter/log/format.go:23]: Should update the comment to match the change.

**Review by blakerouse — APPROVED** [review]
Looks good, one comment.

File: x-pack/elastic-agent/pkg/reporter/log/format.go (hunk 1 above)
Comment (ruflin) [reply x-pack/elastic-agent/pkg/reporter/log/format.go]: updated

**Pull request merged**